x-pack/plugins/search_inference_endpoints/public/components/all_inference_endpoints/tabular_page.tsx

```diff
@@ -0,0 +1,167 @@
+import { EndpointsTable } from './endpoints_table';
+import { TABLE_COLUMNS } from './table_columns';
+
+interface TabularPageProps {
+  addEndpointLabel: string;
+  inferenceEndpoints: InferenceAPIConfigResponse[];
+  setIsInferenceFlyoutVisible: (value: boolean) => void;
+}
+
+const getSortField = (field: string): SortFieldInferenceEndpoint => {
+  if (field in SortFieldInferenceEndpoint) {
+    return SortFieldInferenceEndpoint[field as keyof typeof SortFieldInferenceEndpoint];
+  }
+  return SortFieldInferenceEndpoint.endpoint;
+};
+
+export const TabularPage: React.FC<TabularPageProps> = ({
+  addEndpointLabel,
+  setIsInferenceFlyoutVisible,
+  inferenceEndpoints,
+}) => {
+  const { queryParams, setQueryParams } = useAllInferenceEndpointsState();
+
+  const sorting = useMemo(
+    () => ({
+      sort: {
+        direction: queryParams.sortOrder,
+        field: queryParams.sortField,
+      },
+    }),
+    [queryParams.sortField, queryParams.sortOrder]
+  );
+
+  const pagination = useMemo(
+    () => ({
+      pageIndex: queryParams.page - 1,
+      pageSize: queryParams.perPage,
+      pageSizeOptions: INFERENCE_ENDPOINTS_TABLE_PER_PAGE_VALUES,
+      totalItemCount: inferenceEndpoints.length ?? 0,
+    }),
+    [inferenceEndpoints, queryParams]
+  );
+
+  const tableOnChangeCallback = useCallback(
+    ({ page, sort }) => {
+      let newQueryParams = queryParams;
+      if (sort) {
+        newQueryParams = {
+          ...newQueryParams,
+          sortField: getSortField(sort.field),
+          sortOrder: sort.direction,
+        };
+      }
+      if (page) {
+        newQueryParams = {
+          ...newQueryParams,
+          page: page.index + 1,
+          perPage: page.size,
+        };
+      }
+      setQueryParams(newQueryParams);
+    },
+    [queryParams, setQueryParams]
+  );
+
+  const tableData: InferenceEndpointUI[] = useMemo(() => {
+    return inferenceEndpoints.map((endpoint) => ({
+      endpoint: endpoint.model_id,
+      provider: endpoint.service,
+      type: endpoint.task_type,
+    }));
+  }, [inferenceEndpoints]);
+
+  const sortedTableData: InferenceEndpointUI[] = useMemo(() => {
+    return [...tableData].sort((a, b) => {
+      const aValue = a[queryParams.sortField];
+      const bValue = b[queryParams.sortField];
+
+      if (aValue < bValue) {
+        return queryParams.sortOrder === SortOrder.desc ? 1 : -1;
+      }
+      if (aValue > bValue) {
+        return queryParams.sortOrder === SortOrder.asc ? -1 : 1;
+      }
+      return 0;
+    });
+  }, [tableData, queryParams]);
+
+  const paginatedSortedTableData: InferenceEndpointUI[] = useMemo(() => {
+    const startIndex = pagination.pageIndex * pagination.pageSize;
+    const endIndex = startIndex + pagination.pageSize;
+    return sortedTableData.slice(startIndex, endIndex);
+  }, [sortedTableData, pagination]);
+
+  return (
+    <InferenceEndpointsProvider>
+      <EuiPageTemplate offset={0} restrictWidth={false} grow={false}>
```

Comment:
I don't think this page template should be used here. This component is rendered by `<InferenceEndpoints>` which is rendered by the router in ES3 and by the `<App>` component for stateful.

The PageTemplate should be at the top of the render tree not in the middle. Furthermore for stateful the `<App>` component will be render within a PageTemplate. so we shouldn't render another one below that.

The double rendering for ES3 and Stateful requires us to ensure we get the layout setup correctly for both plugin without conflicts.

Reply:
Moved EuiPageTemplate to the top of the render tree. I modified the layout setup to avoid double rendering.